### PR TITLE
[tests] Force tick on FwdEventTicker

### PR DESCRIPTION
**Pull request by halseth**
Ensures events are properly flushed on `FwdEventTick`. Should also help with the flaky coverage reports.

### CODE REVIEW

**Comment by Roasbeef** [comment]
Is all of this really necessary _just_ to appease a code coverage tool which itself is known to be overly sensitive?

File: htlcswitch/switch_test.go [hunk 1]
```diff
@@ -1835,7 +1836,58 @@ func TestMultiHopPaymentForwardingEvents(t *testing.T) {
...
+	}
+
+	bobLog, ok := n.bobServer.htlcSwitch.cfg.FwdingLog.(*mockForwardingLog)
+	if !ok {
+		t.Fatalf("mockForwardingLog assertion failed")
+	}
+
+	// After sending 5 of the payments, trigger the forwarding ticker, to
+	// make sure the events are properly flushed.
+	bobTicker := n.bobServer.htlcSwitch.cfg.FwdEventTicker.(*ticker.Mock)
```
Comment (wpaulino) [inline htlcswitch/switch_test.go]: Nit: also make type check like we do above?

File: htlcswitch/switch_test.go [hunk 2]
```diff
@@ -1835,7 +1836,58 @@ func TestMultiHopPaymentForwardingEvents(t *testing.T) {
...
+
+	// We'll trigger the ticker, and wait for the events to appear in Bob's
+	// forwarding log.
+	timeout := time.After(15 * time.Second)
+	for {
+
```
Comment (wpaulino) [inline htlcswitch/switch_test.go]: Nit: extra newline.

File: htlcswitch/switch_test.go [hunk 3]
```diff
@@ -1835,7 +1836,58 @@ func TestMultiHopPaymentForwardingEvents(t *testing.T) {
...
+		bobLog.Unlock()
+
+		// Otherwise wait a little bit before checking again.
+		select {
+		case <-time.After(50 * time.Millisecond):
+		case <-timeout:
+			bobLog.Lock()
+			t.Fatalf("expected 5 events in event log, instead "+
+				"found: %v", spew.Sdump(bobLog.events))
+			bobLog.Unlock()
```
Comment (wpaulino) [inline htlcswitch/switch_test.go]: Perhaps this should be deferred? The `t.Fatal` call _shouldn't_ allow it to get to this point.

Comment (halseth) [reply htlcswitch/switch_test.go]: True, good catch :)

**Pull request merged**